debezium-embedded/src/test/java/io/debezium/pipeline/AbstractBlockingSnapshotTest.java

```diff
@@ -246,6 +247,37 @@ public void executeBlockingSnapshotWithEscapedCollectionName() throws Exception
 
     }
 
+    @Test
+    @FixFor("DBZ-8244")
+    public void anErrorDuringBlockingSnapshotShouldLeaveTheConnectorInAGoodState() throws Exception {
```

Comment:
Could we add a test case where connector is restarted right after the snapshot failure?

Reply:
I suppose you are thinking to this https://issues.redhat.com/browse/DBZ-7903. So in that case I think it will try to wrongly do an initial snapshot. I'll fix that in a separate PR.